packages/iov-cosmos/src/decode.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  Address,
+  Algorithm,
+  Amount,
+  ChainId,
+  Fee,
+  FullSignature,
+  Identity,
+  Nonce,
+  PubkeyBundle,
+  PubkeyBytes,
+  SendTransaction,
+  SignatureBytes,
+  SignedTransaction,
+  TokenTicker,
+} from "@iov/bcp";
+import { Encoding } from "@iov/encoding";
+import amino from "@tendermint/amino-js";
+
+import { AminoTx } from "./types";
+
+const { fromBase64 } = Encoding;
+
+export function decodePubkey(pubkey: amino.PubKey): PubkeyBundle {
+  return {
+    algo: Algorithm.Secp256k1,
+    data: fromBase64(pubkey.value) as PubkeyBytes,
+  };
+}
+
+export function decodeSignature(signature: string): SignatureBytes {
+  return fromBase64(signature) as SignatureBytes;
+}
+
+export function decodeFullSignature(signature: amino.StdSignature, nonce: number): FullSignature {
+  return {
+    nonce: nonce as Nonce,
+    pubkey: decodePubkey(signature.pub_key),
+    signature: decodeSignature(signature.signature),
+  };
+}
+
+export function decodeAmount(amount: amino.Coin): Amount {
+  return {
+    fractionalDigits: 9,
```

Comment:
Is this correct? If I understand correctly, Atom has 6 decimal digits (https://github.com/UCRegistry/registry/issues/14)

Reply:
Is there a source for that? I put this in as a placeholder because I didn't find a definitive reference.

Reply:
I found an official blog post: https://blog.cosmos.network/phase-ii-initiated-cosmos-atom-transfers-enabled-by-governance-831a7e555ab6#edeb

Reply:
I found this:

> The smallest unit of ATOM is the uATOM, or the micro ATOM, where 1 atom = 1,000,000 uATOM.

https://blog.cosmos.network/phase-ii-initiated-cosmos-atom-transfers-enabled-by-governance-831a7e555ab6

This indicates that the u is an ASCII approximation of [µ](https://de.wiktionary.org/wiki/%C2%B5).